app/lib/activitypub/activity/announce.rb

```diff
@@ -5,6 +5,8 @@ def perform
     return reject_payload! if delete_arrived_first?(@json['id']) || !related_to_local_activity?
 
     with_redis_lock("announce:#{value_or_id(@object)}") do
+      return reject_payload! if @object.nil?
```

Comment:
I think this should go even earlier, you can avoid the Redis lock.

Reply:
Done, placed it before the redis lock